codegenerator/cli/templates/static/codegen/src/ReorgDetection.res

```diff
@@ -330,39 +330,40 @@ module LastBlockScannedHashes: {
     ->Belt.Result.map(list => list->makeWithDataInternal(~confirmedBlockThreshold))
   }
 
-  let rec rollBackToBlockTimestampLteInternal = (
-    ~blockTimestamp: int,
+  let min = (arrInt: array<int>) => {
+    arrInt->Belt.Array.reduce(None, (current, val) => {
+      switch current {
+      | None => Some(val)
+      | Some(current) => Js.Math.min_int(current, val)->Some
+      }
+    })
+  }
+
+  let rec rollBackToBlockNumberLtInternal = (
+    ~blockNumber: int,
```

Comment:
We now just rollback to a state where all blocks are lower than the block of the first change to an entity on the given chain.